sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java

```diff
@@ -71,8 +76,8 @@ OrchestrationResponse template() {
     final var inputParams = Map.ofEntries(templateVariable.apply("german"));
 
     final var prompt =
-        new OrchestrationPrompt(inputParams)
-            .withTemplate(TemplateConfig.fromMessages(templateMessage));
+        new OrchestrationPrompt(
+            inputParams, config.withTemplate(TemplateConfig.fromMessages(templateMessage)));
 
     return client.chatCompletion(prompt);
```

Comment:
(Question)

Have you considered moving the config argument to `#chatCompletion(` instead of `new OrchestrationPrompt(`?
(This would reduce the object nesting by 1 level)

Reply:
Implemented this now, I think it's probably slightly better